src/models/sensor-gdx-manager.ts

```diff
@@ -0,0 +1,220 @@
+      this.internalConfig = {
+        collection:{ canControl:true, isCollecting:false },
+        columnListTimeStamp: new Date(),
+        columns:{ },
+        currentInterface: "Vernier Go Direct",
+        currentState: "unknown",
+        os: { name: "Fake", version: "1.0.0"},
+        requestTimeStamp: new Date(),
+        server: { arch: "Fake", version: "1.0.0" },
+        sessionDesc: "Fake",
+        sessionID: "1234",
+        sets:{
+          "100": {
+            name: "Run 1",
+            colIDs: [100]
+          }
+        }
+      };
+    }
+
+    sendSensorConfig(includeOnConnect:boolean) {
+      let sensorConfig = new SensorConfiguration(this.internalConfig);
+      if (includeOnConnect) {
+        this.onSensorConnect(sensorConfig);
+      }
+      this.onSensorStatus(sensorConfig);
+    }
+
+    startPolling() {
+      setTimeout(() => {
+        this.sendSensorConfig(true);
+      }, 10);
+      setInterval(() => {
+        //TODO: do we need to cancel while collecting or while disconnected?
+        this.pollSensor();
+      }, 1000);
+
+    }
+
+    pollSensor() {
+      const readLiveData = async () => {
+        if (!this.disconnectRequested) {
+          this.enabledSensors.forEach((sensor: any, index: number) => {
+            const cNum = this.initialColumnNum + index;
+            this.internalConfig.columns[cNum].liveValueTimeStamp = new Date();
+            this.internalConfig.columns[cNum].liveValue = sensor.value.toString();
+          });
+          this.sendSensorConfig(false);
+        }
+      };
+      readLiveData();
+    }
+
+    hasSensorData() {
+      return this.hasData;
+    }
+
+    requestStart() {
+      console.log("Reading GDX measurements");
+
+      let startCollectionTime = Date.now();
+
+      const readData = async () => {
+        this.enabledSensors.forEach((sensor: any, index: number) => {
+          const cNum = this.initialColumnNum + index;
+          const time = Date.now() - startCollectionTime;
+          this.updateSensorValue(cNum.toString(), time / 1000, sensor.value);
+        });
+
+        if (!this.stopRequested) {
+          // Repeat
+          setTimeout(readData, 10);
+        } else {
+          this.onSensorCollectionStopped();
+          this.stopRequested = false;
+        }
+      };
+
+      readData();
+
+    }
+
+    updateSensorValue(ID:string, time:number, value:number) {
+      if (!value) {
+        return;
+      }
+      this.internalConfig.columns[ID].liveValue = value.toString();
+      this.hasData = true;
+
+      this.onSensorStatus(new SensorConfiguration(this.internalConfig));
+      const data:NewSensorData = {};
+      data[ID] = [[time, value]];
+      this.onSensorData(data);
+    }
+
+    requestStop() {
+      this.stopRequested = true;
+    }
+
+    async getBatteryLevel() {
+      const batteryLevel = await this.gdxDevice.getBatteryLevel();
+      console.log("Battery Level: " + batteryLevel);
+      return batteryLevel;
+    }
```

Comment:
We're not using this function at present (not have I seen battery level in the new UI specs), but @scytacki mentioned that it could be useful, and I agree.  We might want to check with Michael and Dan to see if battery level is something that we want to show to the user.

Reply:
It makes sense to keep the function even if we're not currently using it. I agree that it seems like something useful to display to the user in some fashion.

Reply:
Michael actually had a battery level in one of the most recent mock-ups, but we decided to punt on it for now,  I still think this will eventually be a good feature to have.